packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FlutterWebView.m

```diff
@@ -128,154 +128,154 @@ - (void)dealloc {
+    [self onReload:_webView];
+    result(nil);
   } else if ([[call method] isEqualToString:@"currentUrl"]) {
-    [self onCurrentUrl:call result:result];
+    result([self onCurrentUrl:_webView]);
   } else if ([[call method] isEqualToString:@"evaluateJavascript"]) {
-    [self onEvaluateJavaScript:call result:result];
+    [self onEvaluateJavaScript:_webView jsString:call.arguments result:result];
   } else if ([[call method] isEqualToString:@"addJavascriptChannels"]) {
-    [self onAddJavaScriptChannels:call result:result];
+    [self onAddJavaScriptChannels:_webView channelNames:call.arguments];
+    result(nil);
   } else if ([[call method] isEqualToString:@"removeJavascriptChannels"]) {
-    [self onRemoveJavaScriptChannels:call result:result];
+    [self onRemoveJavaScriptChannels:_webView channelNamesToRemove:call.arguments];
+    result(nil);
   } else if ([[call method] isEqualToString:@"clearCache"]) {
     [self clearCache:result];
   } else if ([[call method] isEqualToString:@"getTitle"]) {
-    [self onGetTitle:result];
+    result([self onGetTitle:_webView]);
   } else if ([[call method] isEqualToString:@"scrollTo"]) {
-    [self onScrollTo:call result:result];
+    NSDictionary *arguments = [call arguments];
+    [self onScrollTo:_webView x:arguments[@"x"] y:arguments[@"y"]];
+    result(nil);
   } else if ([[call method] isEqualToString:@"scrollBy"]) {
-    [self onScrollBy:call result:result];
+    NSDictionary *arguments = [call arguments];
+    [self onScrollBy:_webView x:arguments[@"x"] y:arguments[@"y"]];
+    result(nil);
   } else if ([[call method] isEqualToString:@"getScrollX"]) {
-    [self getScrollX:call result:result];
+    result([self getScrollX:_webView]);
   } else if ([[call method] isEqualToString:@"getScrollY"]) {
-    [self getScrollY:call result:result];
+    result([self getScrollY:_webView]);
   } else {
     result(FlutterMethodNotImplemented);
   }
 }
 
-- (void)onUpdateSettings:(FlutterMethodCall*)call result:(FlutterResult)result {
-  NSString* error = [self applySettings:[call arguments]];
-  if (error == nil) {
-    result(nil);
-    return;
-  }
-  result([FlutterError errorWithCode:@"updateSettings_failed" message:error details:nil]);
+- (FlutterError *_Nullable)onUpdateSettings:(WKWebView *)webView
+                                   settings:(NSDictionary<NSString *, id> *)settings {
+  NSString *error = [self applySettings:webView settings:settings];
+  if (error == nil) return nil;
+  return [FlutterError errorWithCode:@"updateSettings_failed" message:error details:nil];
 }
 
-- (void)onLoadUrl:(FlutterMethodCall*)call result:(FlutterResult)result {
-  if (![self loadRequest:[call arguments]]) {
-    result([FlutterError
-        errorWithCode:@"loadUrl_failed"
-              message:@"Failed parsing the URL"
-              details:[NSString stringWithFormat:@"Request was: '%@'", [call arguments]]]);
+- (FlutterError *_Nullable)onLoadUrl:(WKWebView *)webView
```

Comment:
Can be marked static when the `loadRequest:request:` method is also marked static (see comment later).